ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java

```diff
@@ -171,7 +176,37 @@ private ExecuteResult execute(
       final Injector injector) {
     final PreparedStatement<?> prepared = executionContext.prepare(stmt);
     final ConfiguredStatement<?> configured =
-        ConfiguredStatement.of(prepared, overriddenProperties, ksqlConfig);
-    return executionContext.execute(injector.inject(configured));
+        injector.inject(ConfiguredStatement.of(prepared, overriddenProperties, ksqlConfig));
+
+    final CustomExecutor executor =
+        CustomExecutors.CUSTOM_EXECUTORS.getOrDefault(
+            configured.getStatement().getClass(),
+            executionContext::execute);
+
+    return executor.apply(configured);
+  }
+
+  @FunctionalInterface
+  private interface CustomExecutor extends Function<ConfiguredStatement<?>, ExecuteResult> { }
+
+  @SuppressWarnings("unchecked")
+  private static final class CustomExecutors {
```

Comment:
Any reason why this can't be an `enum` like the other `CustomExecutors` and `CustomValidators` class?  Seems much cleaner...

Reply:
changed - though with only two executors the overhead is just a little large